app/vmalert/notifier/init.go

```diff
@@ -25,6 +25,9 @@ var (
 		"Enable this flag if you want vmalert to evaluate alerting rules without sending any notifications to external receivers (eg. alertmanager). "+
 		"-notifier.url, -notifier.config and -notifier.blackhole are mutually exclusive.")
 
+	headers = flagutil.NewArrayString("notifier.headers", "Optional HTTP headers to send with each request to the corresponding -notifier.url. "+
```

Comment:
shall we mention that this flag will be respected only if `notifier.url` is specified?

Reply:
It's mentioned in the description `xxxx to the corresponding -notifier.url`, as other flags like `-notifier.basicAuth.username`.